flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -331,7 +334,8 @@ public void assignExclusiveSegments() throws IOException {
 		}
 	}
 
-	public void setInputChannel(IntermediateResultPartitionID partitionId, InputChannel inputChannel) {
+	@VisibleForTesting
+	void setInputChannel(IntermediateResultPartitionID partitionId, InputChannel inputChannel) {
```

Comment:
add `TODO` to drop it this method or make it work with `InputChannel[] channels;` field?

Reply:
It makes sense to do that in formal codes. For PoC, I did not pay much efforts for the refactoring work. :)

Reply:
Yes, but I meant just to add a `TODO`, so that it doesn't slip through our attention :)

Reply:
I already removed that method after refactoring previous many tests, otherwise it would block the travis green to find other potential issues. :)